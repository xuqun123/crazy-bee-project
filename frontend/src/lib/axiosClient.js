import axios from 'axios'

const axiosClient = axios.create({
  baseURL: process.env.REACT_APP_API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    Authorization: localStorage.getItem('jwt') ? `Bearer ${localStorage.getItem('jwt')}` : '',
  },
})

export default axiosClient
