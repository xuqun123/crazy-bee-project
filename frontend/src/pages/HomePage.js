import * as React from 'react'
import TopSlogan from '../components/TopSlogan'
import NFTCollectionsList from '../components/NFTCollectionsList'

function HomePage() {
  return (
    <>
      <TopSlogan
        header="#1 NFT Marketplace in Australia"
        summary="Do you want to connect your crypto wallet, list your existing NFTs, create and mint your
          own NFTs for free? We have all these features covered for you!"
        mainCTAText="Explore more NFTs"
        mainCTALink="/explore"
        enableSignUpPopup={true}
      />
      <NFTCollectionsList />
    </>
  )
}

export default HomePage
