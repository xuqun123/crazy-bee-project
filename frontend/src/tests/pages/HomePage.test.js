import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import HomePage from '../../pages/HomePage'
import axiosClient from '../../lib/axiosClient'

describe('HomePage', () => {
  const nftCollection = { name: 'a testing nft collection', _id: '1234-5678-abcd' }

  afterEach(() => jest.clearAllMocks())

  describe('get nftCollections data via api', () => {
    beforeEach(() => {
      jest.spyOn(axiosClient, 'get').mockResolvedValue({ data: { data: [nftCollection] } })
    })

    it('render the HomePage view properly with expected elements', async () => {
      const { rerender } = render(<HomePage />)

      const sloganHeader = screen.getByText(/#1 NFT Marketplace in Australia/)
      const exploreButton = screen.getByTestId('main-cta-btn')
      const signupButton = screen.getByTestId('secondary-cta-btn')

      await waitFor(() => {
        expect(sloganHeader).toBeInTheDocument()
      })
      expect(sloganHeader.tagName.toLowerCase()).toBe('h1')
      expect(exploreButton).toBeInTheDocument()
      expect(exploreButton.tagName.toLowerCase()).toBe('button')
      expect(signupButton).toBeInTheDocument()
      expect(signupButton.tagName.toLowerCase()).toBe('button')

      rerender(<HomePage></HomePage>)

      const nftCollectionNameComponent = screen.getByText(nftCollection.name)
      expect(nftCollectionNameComponent).toBeInTheDocument()
      expect(nftCollectionNameComponent.tagName.toLowerCase()).toBe('h2')
    })

    it('snapshot the HomePage view with all expected elements', async () => {
      const view = render(<HomePage />)

      await waitFor(() => {
        expect(screen.getByText(/#1 NFT Marketplace in Australia/)).toBeInTheDocument()
      })
      expect(view).toMatchSnapshot()
    })
  })

  describe('failed to get nftCollections data', () => {
    const error = new Error('something goes wrong')
    let mockConsoleError

    beforeEach(() => {
      jest.spyOn(axiosClient, 'get').mockRejectedValue(error)
      mockConsoleError = jest.spyOn(console, 'error').mockImplementation((error) => error)
    })

    it('render the HomePage view without listing any nftCollections', async () => {
      const { rerender } = render(<HomePage />)

      const sloganHeader = screen.getByText(/#1 NFT Marketplace in Australia/)
      const exploreButton = screen.getByTestId('main-cta-btn')
      const signupButton = screen.getByTestId('secondary-cta-btn')

      await waitFor(() => {
        expect(sloganHeader).toBeInTheDocument()
      })
      expect(sloganHeader.tagName.toLowerCase()).toBe('h1')
      expect(exploreButton).toBeInTheDocument()
      expect(exploreButton.tagName.toLowerCase()).toBe('button')
      expect(signupButton).toBeInTheDocument()
      expect(signupButton.tagName.toLowerCase()).toBe('button')

      rerender(<HomePage></HomePage>)

      expect(mockConsoleError).toHaveBeenCalled()
    })
  })
})
