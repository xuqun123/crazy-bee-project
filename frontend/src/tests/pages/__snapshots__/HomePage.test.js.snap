// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage get nftCollections data via api snapshot the HomePage view with all expected elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-10m0cap"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
          >
            #1 NFT Marketplace in Australia
          </h1>
          <p
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
          >
            Do you want to connect your crypto wallet, list your existing NFTs, create and mint your own NFTs for free? We have all these features covered for you!
          </p>
          <div
            class="css-d1ev6n-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="main-cta-btn"
              tabindex="0"
              type="button"
            >
              Explore more NFTs
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div>
              <button
                aria-label="sign up"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                data-mui-internal-clone-element="true"
                data-testid="signup-trigger"
                tabindex="0"
                type="button"
              >
                Signup Now
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1mknzg7-MuiContainer-root"
        width="lg"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-10m0cap"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
        >
          #1 NFT Marketplace in Australia
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
        >
          Do you want to connect your crypto wallet, list your existing NFTs, create and mint your own NFTs for free? We have all these features covered for you!
        </p>
        <div
          class="css-d1ev6n-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="main-cta-btn"
            tabindex="0"
            type="button"
          >
            Explore more NFTs
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div>
            <button
              aria-label="sign up"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-mui-internal-clone-element="true"
              data-testid="signup-trigger"
              tabindex="0"
              type="button"
            >
              Signup Now
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1mknzg7-MuiContainer-root"
      width="lg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1pkc0kw-MuiSkeleton-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
